src/ColorPicker/Halogen/Component.purs

```diff
@@ -0,0 +1,241 @@
+module ColorPicker.Halogen.Component
+  ( picker
+  , Query(GetValue, SetValue, Commit)
+  , Message(..)
+  , Props
+  , PickerEffects
+  )
+  where
+
+import Prelude
+
+import Color (Color)
+import Color as Color
+import ColorPicker.Halogen.Layout (Layout(..), ChildLayout(..), PickerComponent(..), InputTextValue, LazyColor, PositionUpdate, ValueHistory, mapValueHistory, mkLazyColor)
+import ColorPicker.Halogen.Utils.Drag as Drag
+import Control.Monad.Aff.Class (class MonadAff)
+import DOM.Classy.Event (preventDefault)
+import Data.Array (index, mapWithIndex, nubBy)
+import Data.Either (Either(..), either)
+import Data.Either.Nested as Either
+import Data.Foldable (fold, foldr, for_)
+import Data.Functor.Coproduct.Nested as Coproduct
+import Data.List as List
+import Data.Map (Map, insert, lookup)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (mempty)
+import Data.Traversable (sequence)
+import Halogen (liftEff)
+import Halogen as H
+import Halogen.Component.ChildPath as CP
+import Halogen.HTML as HH
+import Halogen.HTML.Events as HE
+import Halogen.HTML.Properties as HP
+import Halogen.Query.HalogenM (halt)
+import NumberInput.Halogen.Component as Num
+
+type State =
+  { color ∷ ValueHistory LazyColor
+  , inputValues ∷ Map Cursor InputTextValue
+  , props ∷ Props
+  }
+
+data Message = NextChange Color | NotifyChange Color
+
+
+type Props =
+  { layout ∷ Layout
+  }
+
+
+data Query a
+  = SetProps Props a
+  | DragStart PositionUpdate Drag.CursorEvent a
+  | DragMove PositionUpdate Drag.DragEvent a
+  | UpdateCurrentColor Color a
+  | NumberComponentUpdate Cursor (Maybe Number) a
+  | TextComponentUpdate Cursor (String → Maybe Color) String a
+  | TextComponentBlur Cursor a
+  | Commit a
+  | Init a
+  | GetValue (ValueHistory Color → a)
+  | SetValue (ValueHistory Color) a
+
+
+type ChildQuery = Coproduct.Coproduct1 (Num.Query Number)
+type Slot = Either.Either1 Cursor
+
+cpNumComponent ∷ CP.ChildPath (Num.Query Number) ChildQuery Cursor Slot
+cpNumComponent = CP.cp1
+
+
```

Comment:
Inconsistent newlines. I.e. why `ChildQuery` and `Slot` are groupped together, but `HTML` is extracted and have two empty lines before?